Plugins/Onnx/src/MLTrackClassifier.cpp

```diff
@@ -0,0 +1,116 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <cassert>
+#include <stdexcept>
+
+#include "Acts/Plugins/Onnx/MLTrackClassifier.hpp"
+
+// parametrized constructor
+Acts::MLTrackClassifier::MLTrackClassifier(Ort::Env& env,
```

Comment:
I'm thinking if it's possible to have a base class for the ML interface if the constructor and the ` runONNXInference` etc. are the same for different ML implementations (we probably will have a different ML implementation in the future). Then the `MLTrackClassifier` could be inherited from this base interface and it only needs to implement the `predictTrackLabel` function.

Reply:
Yes, the constructor and `runONNXInference` function don't really care about the internal details of the ML model that's being used for inference, so it should be possible to separate this "general" functionality from the prediction function (that has to be somewhat specific to the type of ML model used).